log/logtest/factory_test.go

```diff
@@ -67,6 +67,23 @@ func TestRecordFactoryMultiple(t *testing.T) {
 	assertAttributes(t, attrs, record1)
 }
 
+func assertRecord(t *testing.T, want log.Record, got log.Record) {
+	t.Helper()
+
+	assert.Equal(t, want.Timestamp(), got.Timestamp())
+	assert.Equal(t, want.ObservedTimestamp(), got.ObservedTimestamp())
+	assert.Equal(t, want.Severity(), got.Severity())
+	assert.Equal(t, want.SeverityText(), got.SeverityText())
```

Comment:
AFAIK the assertions messages from testify would not be great here. For instance it would be very hard to distguish the assertions failures between Timestamp and ObservedTimestamp.
Can we just use `t.Errorf` here and hand-crafted failure messages like in `assertBody` and `assertAttributes`?